src/App.svelte

```diff
@@ -26,323 +23,182 @@
 </script>
 
 <main class="app">
-	<header class="app-header">
-		<ui5-shellbar primary-title={shellBarTitle} show-notifications notifications-count="2" on:profile-click={handleProfileClick}>
-			<img class="app-header-logo" slot="logo" src={logo} alt="ui5 orange pheonix logo" />
-			<ui5-shellbar-item icon="palette" text="Theme" on:click={handleThemeSettingsToggle} />
-			<ui5-avatar slot="profile" size="XS" initials="JD" />
-		</ui5-shellbar>
-	</header>
+	<Header />
 
 	<ui5-tabcontainer collapsed>
-		<ui5-tab text="My Todos" />
+		<ui5-tab text="My Todos"></ui5-tab>
 	</ui5-tabcontainer>
 
 	<section class="app-content">
 		<div class="create-todo-wrapper">
-			<ui5-input id="add-input" on:input={handleItemInput} placeholder="Type a task..." />
-			<ui5-date-picker id="date-picker" on:input={handleDateInput} on:change={handleDateInput} format-pattern="dd/MM/yyyy" />
-			<ui5-button id="add-btn" on:click={handleAdd} design="Emphasized"> Add Todo </ui5-button>
+			<ui5-input id="add-input" oninput={handleItemInput} placeholder="Type a task..."></ui5-input>
+			<ui5-date-picker id="date-picker" oninput={handleDateInput} onchange={handleDateInput} format-pattern="dd/MM/yyyy"></ui5-date-picker>
+			<ui5-button type="button" id="add-btn" onclick={handleAdd} design="Emphasized">Add Todo</ui5-button>
 		</div>
 
 		<section class="list-todo-wrapper">
-			<ui5-panel class="list-todos-panel" header-text="Incompleted Tasks" collapsed={!$todos.length || undefined}>
-				<TodoList items={$todos} on:item-edit={handleEdit} on:item-delete={handleRemove} on:selection-change={handleDone} />
+			<ui5-panel class="list-todos-panel" header-text="Incompleted Tasks" collapsed={!undoneTodos.length || undefined}>
+				<TodoList items={undoneTodos} on:item-edit={handleEdit} on:item-delete={handleRemove} on:selection-change={handleToggleDone} />
 			</ui5-panel>
 
-			<ui5-panel class="list-todos-panel" header-text="Completed Tasks" collapsed={!$todos.length || undefined}>
-				<TodoList items={$doneTodos} on:item-edit={handleEdit} on:item-delete={handleRemove} on:selection-change={handleUndone} />
+			<ui5-panel class="list-todos-panel" header-text="Completed Tasks" collapsed={!doneTodos.length || undefined}>
+				<TodoList items={doneTodos} on:item-edit={handleEdit} on:item-delete={handleRemove} on:selection-change={handleToggleDone} />
 			</ui5-panel>
 		</section>
 	</section>
 
-	<ui5-dialog bind:this={dialog} header-text={dialogHeaderText}>
+	<ui5-dialog bind:this={dialog} header-text="Edit Todo">
 		<div class="dialog-content">
 			<div class="edit-wrapper">
 				<ui5-label>Title:</ui5-label>
-				<ui5-textarea class="title-textarea" show-exceeded-text maxlength="24" bind:this={dialogTextArea} value={itemEditText} />
+				<ui5-textarea class="title-textarea" show-exceeded-text maxlength={24} value={dialogFields.text} onchange={(event: any) => (dialogFields.text = event.target.value)}></ui5-textarea>
```

Comment:
could not find types for the onchange event
https://ui5.github.io/webcomponents/components/TextArea/#change